torch/_dynamo/guards.py

```diff
@@ -771,14 +780,17 @@ def direct_negation(a, b):
         closure_vars.update(CLOSURE_VARS)
         py_code = f"""\
 def ___make_guard_fn({','.join(closure_vars.keys())}):
-    return lambda {args}: {code}
+    return (lambda {args}: {code})
 """
         if os.environ.get("TORCHDYNAMO_PRINT_GUARDS", None) == "1":
             print("GUARDS", code)
         set_guard_fail_hook(guard_fail_hook)
         out = dict()
-        # print("RUNNING PY CODE", py_code)
-        exec(py_code, global_builder.scope, out)
+        try:
+            exec(py_code, global_builder.scope, out)
+        except:
+            logging.error(f"Code that failed to compile:\n{py_code}")
```

Comment:
@voznesenskym instead of commented out prints, extra logging on exception lets everyone benefit automatically :)